### PR TITLE
Enable cross-chain NFT transfers on asset-hub

### PR DESCRIPTION
**Pull request by Szegoo**
This PR introduces the following changes:

- [x] Adds a `UniquesTransactor` to asset-hub-rococo
- [x] Adds a `UniquesTransactor` to asset-hub-westend

We can't add a transactor for `pallet-nfts` like we do for `pallet-uniques` because `pallet-nfts` uses `nonfungibles_v2::Mutate` instead of `nonfungibles::Mutate`, and making that work would be out of scope of this PR.

With these modifications, reserve-based NFT cross-chain transfers can be performed on asset-hub.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4795595

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4795740

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4796166

File: cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs [hunk 1]
```diff
@@ -134,6 +136,25 @@ pub type FungiblesTransactor = FungiblesAdapter<
...
+	// Use this non-fungibles implementation:
+	Uniques,
+	// This adapter will handle any non-fungible asset from the uniques pallet.
+	UniquesConvertedConcreteId,
+	// Convert an XCM MultiLocation into a local account id:
+	LocationToAccountId,
+	// Our chain's account ID type (we can't get away without mentioning it explicitly):
+	AccountId,
+	// We don't allow teleports.
+	NoChecking,
```
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs:153]:
```suggestion
	// Does not check teleports.
	NoChecking,
```

File: cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs [hunk 2]
```diff
@@ -128,6 +131,25 @@ pub type FungiblesTransactor = FungiblesAdapter<
...
+	// Use this non-fungibles implementation:
+	Uniques,
+	// This adapter will handle any non-fungible asset from the uniques pallet.
+	UniquesConvertedConcreteId,
+	// Convert an XCM MultiLocation into a local account id:
+	LocationToAccountId,
+	// Our chain's account ID type (we can't get away without mentioning it explicitly):
+	AccountId,
+	// We don't allow teleports.
+	NoChecking,
```
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs:148]:
```suggestion
	// Does not check teleports.
	NoChecking,
```

File: cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs [hunk 3]
```diff
@@ -134,6 +136,25 @@ pub type FungiblesTransactor = FungiblesAdapter<
 	CheckingAccount,
 >;
 
+pub type UniquesConvertedConcreteId =
```
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs:139]: pub type should have docs

File: cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs [hunk 4]
```diff
@@ -128,6 +131,25 @@ pub type FungiblesTransactor = FungiblesAdapter<
 	CheckingAccount,
 >;
 
+pub type UniquesConvertedConcreteId =
```
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs:134]: docs

**Review by joepetrowski — APPROVED** [review]

File: cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs [hunk 5]
```diff
@@ -187,8 +209,13 @@ pub type PoolFungiblesTransactor = FungiblesAdapter<
 >;
 
 /// Means for transacting assets on this chain.
-pub type AssetTransactors =
-	(CurrencyTransactor, FungiblesTransactor, ForeignFungiblesTransactor, PoolFungiblesTransactor);
+pub type AssetTransactors = (
+	CurrencyTransactor,
+	FungiblesTransactor,
+	UniquesTransactor,
```
Comment (bkontur) [inline cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs:215]: I would maybe put it at the end, if we expect more fungible than non-fungible operations

Comment (Szegoo) [reply cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs]: Seems like a realistic expectation. Updated the order.

**Comment by Szegoo** [comment]
Not sure why the prdoc CI check is failing.

**Review by franciscoaguirre — APPROVED** [review]

**Pull request merged**